Dockerfile

```
@@ -0,0 +1,13 @@
+FROM node:10
+
+WORKDIR /src
+RUN chown -R node:node /src
+
+USER node
+
+ADD package.json /src/
+ADD package-lock.json /src/
+
+RUN npm install --unsafe-perm
+
+ADD . /src/
```

Best practice is to use the CMD sytnax, https://docs.docker.com/engine/reference/builder/#cmd
E.g. https://github.com/zooniverse/sugar/blob/650dc5ac8fe0442d2bb3625dbbcf4642f2a88460/Dockerfile#L15

this avoid having to modify the entrypoint & cmd manually in other places (e.g. the docker-compose file
```suggestion
ADD . /src/

CMD ["npm", "start"]
```

CMD syntax allows super easy modification via runtime if desired, e.g. 
`docker-compose run -T --rm app npm test` or 
`docker-compose run --rm app bash`

---

We don't want to start a local development server during the Jenkins builds.

---

i don't think it will by default, i believe the jenkins `inside` cmd will override the CMD syntax by default (can't find any docs on this but i think that's how it currently works)
https://github.com/zooniverse/Panoptes-Front-End/blob/63fe2e43b2ed1c795666610375b6adc3ccf282b2/Jenkinsfile#L22-L26

the node dockerfile CMD syntax (default) starts a node repl without any modification to entry point / cmd.
https://github.com/nodejs/docker-node/blob/c2604466d06ba562fd9040d18c57af16545c6a5b/10/alpine3.11/Dockerfile#L100

---

That's really useful to know. I'd been assuming that `docker.build(imageName)` ran the command in the Dockerfile. I'm inclined to try it here, then copy across to PFE so that builds are consistent.